Core/include/Acts/Vertexing/AdaptiveMultiVertexFinder.ipp

```diff
@@ -33,44 +33,44 @@ auto Acts::AdaptiveMultiVertexFinder<vfitter_t, sfinder_t>::find(
 
   int iteration = 0;
   std::vector<const InputTrack_t*> removedSeedTracks;
-  while (((m_cfg.addSingleTrackVertices && !seedTracks.empty()) ||
-          ((!m_cfg.addSingleTrackVertices) && !seedTracks.empty())) &&
-         iteration < m_cfg.maxIterations) {
-    // Tracks that are used for searching compatible tracks
-    // near a vertex candidate
-    std::vector<const InputTrack_t*> searchTracks;
-    if (m_cfg.doRealMultiVertex) {
-      searchTracks = origTracks;
-    } else {
-      searchTracks = seedTracks;
-    }
+  while (!seedTracks.empty() && iteration < m_cfg.maxIterations) {
```

Comment:
So the previous code was obviously nonsensical, but maybe it would be better to replace it with something like
```suggestion
while (((m_cfg.addSingleTrackVertices && !seedTracks.empty()) ||
          (seedTracks.size()>1)) &&
          iteration < m_cfg.maxIterations) {
```

Reply:
Otherwise we should remove the member variable `addSingleTrackVertices`, but I don't know if anyone ever sets this to true

Reply:
right I also thought about fixing this but was a bit scared about performance changes. we can try

Reply:
Alternatively, we could leave this for a separate PR

Reply:
pulled this into https://github.com/acts-project/acts/pull/2931